### DISCUSSION TITLE
Hydration mismatch when using offlineExchange and ssrExchange

### DISCUSSION THREAD

**Discussion by y15e** [urql-graphql/urql#2591] [Q&A] [answered]
I'm trying to use `offlineExchange` and `ssrExchange` with Vue3, but I'm getting "Hydration text content mismatch" warning.
The issue can be reproduced with the following example which pulls random quotes from [demotivational-quotes-api](https://demotivation-quotes-api.herokuapp.com/graphql) public endpoint.

Example page: https://urql-vue-ssr.herokuapp.com/  (Please reload the page several times to trigger the issue.)
Souce code: https://github.com/y15e/urql-vue-ssr

- Hydration mismatch warning in browser console
![DeepinScreenshot_google-chrome_20220807134937](https://user-images.githubusercontent.com/27120451/183277098-e026475c-2248-4aed-afe0-fef5f1cae322.png)

`ssr.extractData()` returns **id:16** for client side hydration, but it seems **id:25** is loaded from existing Graphcache. 
How can I fix the hydration mismatch? Thank you.

- ssr.extractData()
![DeepinScreenshot_google-chrome_20220807135049](https://user-images.githubusercontent.com/27120451/183277097-3a1fbf34-35fe-4ed5-9555-0dbd8a23d4a8.png)

- urql Graphcache
![DeepinScreenshot_google-chrome_20220807135115](https://user-images.githubusercontent.com/27120451/183277100-d8777654-125c-49b1-95da-05e6ead82fb6.png)

**Comment by kitten** [comment, marked as answer]
We have unfortunately not implemented a solution yet for the `ssrExchange` and the offline-capable `offlineExchange` to work together.

We have an issue for this, but I believe we haven't gotten to it yet. Sorry! https://github.com/FormidableLabs/urql/issues/1512

Essentially, the problem is that the `offlineExchange` blocks operations while it's rehydrating the cache. This is why it isn't able to use the SSR results, since it attempts to rely on its result from the storage only.
The solution we came up with was to fall back to sending operations on to the next exchange when it's still waiting for the storage, which would enable to SSR exchange to give it a result.

**Reply by y15e** [reply]
Thank you so much for your prompt reply! I will track #1512.
